pkg/apis/core/types.go

```diff
@@ -3406,8 +3406,23 @@ type LoadBalancerIngress struct {
 	// (typically AWS load-balancers)
 	// +optional
 	Hostname string
+
+	// RouteType specifies the type of route to use for this ingress
+	// Defaults to `VIP`
```

Comment:
this has to be defaulted here?
https://github.com/kubernetes/kubernetes/blob/70302466f4deeef5838cc420e770b2f9e2fe1425/pkg/apis/core/v1/defaults.go#L99

Reply:
Since it needs to get populated by the cloud controller manager, I'm not sure it's the right place. Either we don't default it or we loop through the `ingress` once `EnsureLoadBalancer` is called. (Not sure when the `SetDefaults_Service` is called exactly)

Reply:
@thockin and/or @andrewsykim  would know better,
 can you please advice here?